cmd/installer/pkg/install/install.go

```
@@ -68,7 +76,15 @@ func (m Mode) IsImage() bool {
 }
 
 // Install installs Talos.
-func Install(ctx context.Context, p runtime.Platform, mode Mode, opts *Options) (err error) {
+//
+//nolint:gocyclo
+func Install(ctx context.Context, p runtime.Platform, mode Mode, opts *Options) error {
+	if c := procfs.ProcCmdline().Get(constants.KernelParamBoard).First(); c != nil {
+		if *c != constants.BoardNone && quirks.New(version.Tag).SupportsOverlay() {
+			return fmt.Errorf("using standard installer image is not supported for board: %s, use an installer with overlay", *c)
```

when upgrading from Talos 1.6 to Talos 1.7 with overlay, the board parameter will be present (from 1.6 Talos), so if we fail hard here, we'll block the upgrade path

---

ahh I forgot `version.Tag` comes from `installer` and not from the one on disk, i think we still need to fail early rather than make the board unbootable